packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -505,6 +508,12 @@ [.. primaryCtorOrderedParams.Select(p => p.InitializationValue ?? p)
                 this);
         }
 
+        public ScmMethodProviderCollection GetMethodCollectionByOperation(InputOperation operation)
```

Comment:
This exposed method can replace https://github.com/Azure/azure-sdk-for-net/blob/2c259e361300b2aa42c92db355ed874f676fc947/eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs#L433-L434